internal/globalconfig/globalconfig.go

```diff
@@ -36,3 +37,17 @@ func SetAnalyticsRate(rate float64) {
 	cfg.analyticsRate = rate
 	cfg.mu.Unlock()
 }
+
+// ServiceName returns the global service name set for this application.
+func ServiceName() string {
+	cfg.mu.RLock()
+	defer cfg.mu.RUnlock()
+	return cfg.serviceName
+}
+
+// SetServiceName sets the global service name set for this application.
+func SetServiceName(name string) {
```

Comment:
Yep this API makes sense to me.

Reply:
Yeah this isn't great but it's the best we can do. Because it's `internal` it's slightly ok :)